lib/posix.lua.in

```diff
@@ -519,5 +519,27 @@ if _DEBUG ~= false then
   end
 end
 
+local function prepare_args (pat_mark)
+  local pat, mark = "*", 0
+  if type (pat_mark) == "table" then
+    if type (pat_mark.pattern) == "string" then
+      pat = pat_mark.pattern
+    end
+    if pat_mark.MARK then
+      mark = M.GLOB_MARK
+    end
+  elseif type (pat_mark) == "string" then
+    pat = pat_mark
+  end
+
+  return pat, mark
+end
+
+local cglob = M.glob
+
+M.glob = function (...)
+  local pat, mark = prepare_args(...)
+  return cglob(pat, mark)
+end
 
```

Comment:
To follow the style of the rest of the functions in this file, we'd then wrap the function above in argument checking if `_DEBUG ~= false`.